### PR TITLE
Add configuration to run MAVSDK on the autopilot side

### PR DESCRIPTION
**Pull request by JonasVautherin**
This adds a way for MAVSDK to advertise itself as an autopilot.

@julianoes: I hardcoded the system id to 1. Is that a valid first implementation, or should I look into having it as a parameter somewhere?

### CODE REVIEW

File: src/core/mavsdk_impl.cpp [hunk 1]
```diff
@@ -345,10 +351,15 @@ uint8_t MavsdkImpl::get_own_component_id() const
 uint8_t MavsdkImpl::get_mav_type() const
 {
     switch (_configuration.load()) {
+        case Mavsdk::Configuration::Autopilot:
+            return MAV_TYPE_GENERIC;
```
Comment (julianoes) [inline src/core/mavsdk_impl.cpp]: I wonder what that means.

Comment (JonasVautherin) [reply src/core/mavsdk_impl.cpp]: "Generic micro air vehicle": https://mavlink.io/en/messages/common.html#MAV_TYPE_GENERIC

:sweat_smile:

**Comment by hamishwillee** [comment]
> I hardcoded the system id to 1. Is that a valid first implementation, or should I look into having it as a parameter somewhere?

FWIW IMO it is a valid first implementation.

**Comment by JonasVautherin** [comment]
Ok, merging for now, that can be improved later.

Regarding the QGC filters, that's documented here: https://github.com/mavlink/MAVSDK/issues/917. Makes sense to me to solve that in a different PR.

**Pull request merged**